### ISSUE TITLE
Permission denied in terminal (Lollipop)

### ISSUE THREAD

**Issue by Mu99u** [kosborn/p2p-adb#35]
Hi Kos, hats off to ur efforts really nice project..I have moto g 1st gen with lollipop.. I tried the method as in terminal I typed su then #/sdcard/p2p-adb-master/run.sh but it shows permission denied :( ..plz help..

**Comment by denzuko** [comment]
Did you remember to [set the executable bit](http://linux.die.net/man/1/chmod) on `run.sh`?

**Comment by duramato** [comment]
@Mu99u  & @denzuko  i found a solution ,if you copy it to the root of our phone like /data/local (which was what i used) it works but if i leave it on the sdcard i get permission denied

**Comment by denzuko** [comment]
@duramato good find on the work around. but what caused this? Is the sdcard being mounted noexec? What about file permissions and acl policies?

**Comment by duramato** [comment]
@denzuko yes i believe its from the `noexec` part that blocks any execution of scripts ,etc from there

**Comment by denzuko** [comment]
Well if that's the case, then wouldn't a better work around be to execute `mount -o remount,exec /sdcard`?

**Comment by duramato** [comment]
that does seem to work on my device i believe the sdcard is emulated

**Comment by denzuko** [comment]
@kosborn Looks like we can close this one as resolved
